### ISSUE TITLE
The wx.EXPAND causes wx._core.wxAssertionError - when I remove wx.EXPAND, program runs, but not to my appearance

### ISSUE THREAD

**Issue by AnthonyTheKoala** [wxFormBuilder/wxFormBuilder#645]
There appears to be a problem when you use wx.EXPAND.

I have been learning how to use the wxFormBuilder and wanted to add a statictext, combo and button.  I was learning from a tutorial and all the controls were to be as wide as the 'form'. The wxFormBuilder added the wx.EXPAND to make the controls as wide as the underlying form.

Snippet of the affected code - note the wx.EXPAND generated by the wxFormBuilder
<pre>
bSizer1.Add( self.HelloText, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|**wx.EXPAND**, 5 )
bSizer1.Add( self.choice, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|**wx.EXPAND,** 5 )
bSizer1.Add( self.button, 0, **wx.EXPAND|**wx.ALIGN_CENTER_HORIZONTAL, 5 )
</pre>

When I run the program, gui.py
<pre>
c:/python38/python.exe gui.py
</pre>

I get the following runtime error
<pre>
c:/python38/python.exe gui.py
Traceback (most recent call last):
  File "gui.py", line 63, in <module>
    frame = MainFrame(None).Show()
  File "gui.py", line 31, in __init__
    bSizer1.Add( self.HelloText, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL|wx.EXPAND, 5 )
wx._core.wxAssertionError: C++ assertion "!(flags & (wxALIGN_RIGHT | wxALIGN_CENTRE_HORIZONTAL))" failed at ..\..\src\common\sizer.cpp(2089) in wxBoxSizer::DoInsert(): Horizontal alignment flags are ignored with wxEXPAND
</pre>

Snippet of the code without - wx.EXPAND
<pre>
bSizer1.Add( self.HelloText, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL, 5 )
bSizer1.Add( self.choice, 0, wx.ALL|wx.ALIGN_CENTER_HORIZONTAL, 5 )
bSizer1.Add( self.button, 0, wx.ALIGN_CENTER_HORIZONTAL, 5 )
</pre>

When I run the program with the wx.EXPAND removed everything runs, BUT the controls static, combo and button are not expanded to the width of the underlying form.

In other words, the wx.EXPAND caused the error, 

Any idea please?
OS: Win 7
Python: 3.8.6
<pre>
import wx
wx.__version__
'4.1.0'
</pre>

Thank you,
Anthony of Sydney

**Comment by sodevel** [comment]
You have posted the answer to your question yourself:

> wx._core.wxAssertionError: C++ assertion "!(flags & (wxALIGN_RIGHT | wxALIGN_CENTRE_HORIZONTAL))" failed at ..\..\src\common\sizer.cpp(2089) in wxBoxSizer::DoInsert(): Horizontal alignment flags are ignored with wxEXPAND

Interesting that this assertion causes a program termination in Python, in C++ it displays this message in debug mode and does nothing in release mode. wxFB cannot validate all your settings, it is your responsibility to not use invalid settings combinations. Check the wxWidgets documentation about how the different sizers work and what restrictions apply to the sizer flags, especially for wxEXPAND there are differences between 1D sizers and 2D sizers.

**Comment by AnthonyTheKoala** [comment]
Dear Steffen,
Thank you for the reply.
First I was using the wxFormBuilder to provide the 'skeleton' code in Python. The error message came from using the python-generated code in the python interpreter.

I was learning the code from a tutorial on wxFormBuilder at https://www.academia.edu/21204860/wxFormBuilder_Tutorial_on_A_GUI_for_making_GUIs_for_Python . I followed the tutorial exactly as presented. The tutorial worked for the author. The code generated from the wxFormBuilder came from the wxFormBuilder generated code.

In the tutorial each of the wxStaticText, wxChoice and wxButton controls were expanded by either pressing the 'Expand' icon **OR** pressing Alt +W.

Why it did not work for me unless I removed the wx.Expand code. The expansion was set in the wxFormBuilder IDE by clicking on the property of each control.

I didn't have the choice to have the C++ compiler run in debug mode when the code was written in python. There was no way I could adjust debug or release from wxFormBuilder.

Thus when I posted the problem, I unfortunately did not answer the question I asked because I have no idea of C++ coding.

Here is the information:
Python: v3.8.5
wxPython: v4.1.0
wxFormBuilder: v3.9.0
msys64:
mingw32's gcc: v10.2.0
mingw64's gcc : v10.2.0
OS: Win 7

I thank you for your time,
Anthony of Sydney

**Comment by maimoy** [comment]
I followed the same tutorial and have the same error and after a lot o searching i found that ininstall the latests wxPython                           4.1.1 and installing the wxPython 4.0.7.post2 this problem with wxExpand resolved. Some corrections needed by the devs in the next releases

**Comment by sodevel** [comment]
No, you didn't solve any problem, you just downgraded to a version that does not check for that specific error so your code doesn't trigger that assertion. That doesn't fix your code, it's still broken. Fix your code to not use invalid flag combinations.
